File: javascript/syntax/9.functions.js
// Functions

// Function with parameter
function myFunction(fname) {
    console.log(fname + " Refsnes");
}

// Function with multiple parameters
function myFunctionWithParams(fname, lname) {
    console.log(fname + " " + lname);
}

// Function with default parameter
function myFunctionWithDefault(fname, lname = "Refsnes") {
    console.log(fname + " " + lname);
}

// Arrow function with parameter
const myArrowFunction = (fname) => {
    console.log(fname + " Refsnes");
};

// Test the functions
myFunction("John");
myFunctionWithParams("John", "Doe");
myFunctionWithDefault("Jane");
myArrowFunction("Alice"); 

File: java/syntax/9.Functions.java
// Functions
public class Functions {
    // Function with parameter
    public static void myFunction(String fname) {
        System.out.println(fname + " Refsnes");
    }

    // Function with multiple parameters
    public static void myFunction(String fname, String lname) {
        System.out.println(fname + " " + lname);
    }

    // Function with default parameter (Java doesn't have default parameters, using overloaded methods)
    public static void myFunction(String fname, String lname) {
        System.out.println(fname + " " + lname);
    }

    // Overloaded method with default parameter
    public static void myFunction(String fname) {
        myFunction(fname, "Refsnes");
    }

    public static void main(String[] args) {
        // Test the functions
        myFunction("John");
        myFunction("John", "Doe");
        myFunction("Jane", "Smith");
    }
} 